numpy/core/__init__.py

```diff
@@ -3,9 +3,34 @@
 from .info import __doc__
 from numpy.version import version as __version__
 
+import os
+
+# on Windows NumPy loads an important OpenBLAS-related DLL
+# and the code below aims to alleviate issues with DLL
+# path resolution portability with an absolute path DLL load
+if os.name == 'nt':
+    from ctypes import WinDLL
+    import glob
+    # convention for storing / loading the DLL from
+    # numpy/.libs/, if present
+    libs_path = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                             '..', '.libs'))
+    DLL_filenames = []
+    if os.path.isdir(libs_path):
+        for filename in glob.glob(os.path.join(libs_path, '*openblas*dll')):
+            # NOTE: would it change behavior to load ALL
+            # DLLs at this path vs. the name restriction?
+            WinDLL(os.path.abspath(filename))
+            DLL_filenames.append(filename)
+    if len(DLL_filenames) > 1:
+        import warnings
+        warnings.warn("loaded more than 1 DLL from .libs:",
+                      stacklevel=1)
+        for DLL_filename in DLL_filenames:
+            warnings.warn(DLL_filename, stacklevel=1)
```

Comment:
Emitting multiple warnings like this is a bad idea - it's likely that the default once role causes only the first item to be shown. Please build the entire string first.

Reply:
Ok, I revised to use a single string / warning, with DLL filenames interleaved between newline chars.